lib/puppet/provider/foreman_host/rest_v3.rb

```diff
@@ -2,7 +2,7 @@
   confine :feature => [:json, :oauth]
 
   def exists?
-    !id.nil?
+    !id.nil? && !resource[:force]
```

Comment:
I'd swap it since `id` ends up doing the HTTP call while `resource[:force]` is just reading a property.
```suggestion
    !resource[:force] && !id.nil?
```